### PR TITLE
Make collector compatible with ceph luminous

### PR DESCRIPTION
**Pull request by brtkwr**

### CODE REVIEW

**Review by dougszumski — COMMENTED** [review]
Are we able to come up with something which allows the old behaviour to work, but switches to the new behaviour only for our situation? Ideally, we want to submit this upstream and we don't want to break anyone else's workflow. What do you think?

**Comment by brtkwr** [comment]
Okay it should be compatible with earlier versions of ceph now.

**Comment by brtkwr** [comment]
@dougszumski can I merge this branch?

File: monasca_agent/collector/checks_d/ceph.py [hunk 1]
```diff
@@ -51,6 +52,10 @@ def check(self, instance):
         self.CLUSTER = instance.get('cluster_name', 'ceph')
         self.dimensions = self._set_dimensions({'ceph_cluster': self.CLUSTER,
                                                 'service': 'ceph'}, instance)
+        match_version = re.search(
+                            _CEPH_VERSION_REGEX,
+                            self._ceph_cmd('version'))
+        self.preluminous = match_version.group(1).split('.') < ['12','2','0']
```
Comment (dougszumski) [inline monasca_agent/collector/checks_d/ceph.py]: How about using something like the snippet below to better describe what '12','2','0' is, and also make the version parsing more robust?

```
>> from pkg_resources import parse_version
>> _CEPH_LUMINOUS_RELEASE_VERSION = '12.2.0'
>> parse_version('12.2.1.dev') > parse_version(_CEPH_LUMINOUS_RELEASE_VERSION)
>> True
```

File: monasca_agent/collector/checks_d/ceph.py [hunk 2]
```diff
@@ -359,10 +365,16 @@ def _get_stats_metrics(self, ceph_status, ceph_status_plain):
...
+            health_key = 'overall_status'
+        else:
+            health_key = 'status'
+
         metrics['ceph.cluster.health_status'] = self._parse_ceph_status(
-            ceph_status_health['overall_status'])
+            ceph_status_health[health_key])
 
-        for s in ceph_status_health['summary']:
+        for s in ceph_status_health.get('summary', []):
```
Comment (dougszumski) [inline monasca_agent/collector/checks_d/ceph.py:385]: Has the summary field actually been removed or has it been renamed?

File: monasca_agent/collector/checks_d/ceph.py [hunk 3]
```diff
@@ -416,8 +428,8 @@ def _get_mon_metrics(self, ceph_status):
         {'monitor1': {metric1': value1, ...}, 'monitor2': {metric1': value1}}
         """
         mon_metrics = {}
-        for health_service in ceph_status['health']['health'][
-                'health_services']:
+        for health_service in ceph_status.get('health').get('health', {}).get(
```
Comment (dougszumski) [inline monasca_agent/collector/checks_d/ceph.py:439]: Is this information available, but in a different place?

**Review by dougszumski — CHANGES_REQUESTED** [review]
Thanks @brtknr, this gets things going. It would be nice if we could propose it upstream when you're finished so we can avoid deploying our own fork of the monasca agent.

Please could you update the tests (https://github.com/openstack/monasca-agent/blob/master/tests/checks_d/test_ceph.py). It would be nice to see example output from the new ceph luminous command in there to make it clear whether some output is no longer available, or is available but under a different field.

File: monasca_agent/collector/checks_d/ceph.py (hunk 3 above)
Comment (brtkwr) [reply monasca_agent/collector/checks_d/ceph.py:439]: Hmm I don't actually know, I was just trying to get it to work at the time without coughing errors iirc.

Comment (brtkwr) [reply monasca_agent/collector/checks_d/ceph.py:439]: I suppose i could check the monasca agent log on soft iron ceph cluster

**Comment by brtkwr** [comment]
Okay makes sense to close this then.

**Review by dougszumski — APPROVED** [review]
Changes look good, still a couple of old comments which could be nice to investigate, but we could merge this now it works and follow up later.

I wonder if having a separate plugin for the various Ceph releases would be simpler to maintain in the longer run, particularly as we don't have tests for the various releases.

**Pull request merged**